src/main/java/codeu/model/data/Event.java

```diff
@@ -0,0 +1,173 @@
+   */
+	public Event(Conversation conversation) {
+      userStore = UserStore.getInstance();
+      conversationStore = ConversationStore.getInstance();
+
+      titleOfConversation = conversation.getTitle();
+      authorIdForConversation = conversation.getOwnerId();
+      conversationCreationTime = conversation.getCreationTime();
+      eventType = 'c';
+
+      messageCreationTime = null;
+      authorIdForMessage = null;
+      conversationTitleOfMessage = null;
+      messageContent = null;
+
+      userCreationTime = null;
+      nameOfUser = null;
+
+	}
+
+  /**
+   * Constructs a new event using a Message object.
+   *
+   * @param message an object of type Message
+   */
+	public Event(Message message) {
+      userStore = UserStore.getInstance();
+      conversationStore = ConversationStore.getInstance();
+
+      titleOfConversation = null;
+      authorIdForConversation = null;
+      conversationCreationTime = null;
+
+      messageCreationTime = message.getCreationTime();
+      authorIdForMessage = message.getAuthorId();
+      conversationTitleOfMessage = conversationStore.getConversationWithId(message.getConversationId()).getTitle(); //null error is here
+      messageContent = message.getContent();
+      eventType = 'm';
+
+      userCreationTime = null;
+      nameOfUser = null;
+	}
+
+  /**
+   * Constructs an new event using a User object.
+   *
+   * @param user an object of type User
+   */
+    public Event(User user) {
+      userStore = UserStore.getInstance();
+      conversationStore = ConversationStore.getInstance();
+
+      titleOfConversation = null;
+      authorIdForConversation = null;
+      conversationCreationTime = null;
+
+      messageCreationTime = null;
+      authorIdForMessage = null;
+      conversationTitleOfMessage = null;
+      messageContent = null;
+
+      userCreationTime = user.getCreationTime();
+      nameOfUser = user.getName();
+      eventType = 'u';
+	}
+
+	/** Outputs a string based on the type given in the constructor. */
+	public String toString() {
+
+	  if(eventType == 'c') {
+        return dateTime_ToString(conversationCreationTime) + ": "+ 
+          userStore.getUser(authorIdForConversation).getName() + 
+          " created a new conversation: " + titleOfConversation;
+	  }
+	  if(eventType == 'm') {
+        return dateTime_ToString(messageCreationTime) + " PST: " + 
+          userStore.getUser(authorIdForMessage).getName() + 
+          " sent a message in " + conversationTitleOfMessage + 
+          ": " + "\"" + messageContent + "\"";
+	  }
+	  if(eventType == 'u') {
+	  	return dateTime_ToString(userCreationTime) + " PST: " + nameOfUser + " joined!";
+	  }
+	  else{
+	  	System.err.println("This object has no event type. This is impossible. I don't know how you even caused this error");
+	  	return null;
+	  }
+	}
+
+    /** Formats time of type Instant into a string. */
+	public String dateTime_ToString(Instant unformattedTime) {
```

Comment:
Either
dateTimeToString()
or
simply toString(), it  takes an Instant as parameter, so it is pretty obvious it is converting a time to string.